OpenRA.Mods.Common/Traits/Replacement.cs

```diff
@@ -16,8 +16,8 @@ namespace OpenRA.Mods.Common.Traits
 	public class ReplacementInfo : TraitInfo<Replacement>
 	{
 		[FieldLoader.Require]
-		[Desc("Replaceable Types this Actor replaces.")]
-		public readonly string[] Replaces = { };
+		[Desc("Replacement type (matched against Conditions in Replaceable.")]
+		public readonly string Type = null;
```

Comment:
And then keeping this as an array (actually now that I think about it this should be a `HashSet<string>`) too.  You can then use `Overlaps` to check whether the two sets overlap.